faridafaqiri/pokemon-search-app
# Pokémon Search App

## Description:
This pull request introduces significant updates and features to the Pokémon Search App. Below are the key additions and changes:

1. HTML Structure:
   Added basic structure with a search input, search button, and a detailed view section for displaying Pokemon
   information.
  Included a table to display base stats for the searched Pokémon.
  Utilized semantic HTML elements for better accessibility and readability.
2. CSS Styling:
   Added custom styles for the app, including background images, font styles, and responsive design.
   Implemented specific color coding for different Pokémon types to enhance visual clarity.
   Ensured the layout is responsive and adjusts well to different screen sizes with media queries.
3. JavaScript Functionality:
   Implemented search functionality to fetch Pokémon data from the PokeAPI using fetch API.
   Displayed fetched Pokémon details, including name, ID, image, weight, height, types, and base stats.
   Added error handling to alert the user if the Pokémon is not found.
   Used event listeners to handle search input and button click, allowing search on pressing "Enter".

Screenshots:
Screenshots of the updated app showcasing the search functionality, detailed view, and type colors.


![Capture](https://github.com/faridafaqiri/pokemon-search-app/assets/77193878/217f31e2-e28e-4fb4-8b31-bf411c9a093c)
![Capture2](https://github.com/faridafaqiri/pokemon-search-app/assets/77193878/3e704421-17f6-4336-8fc9-a27ccac52b74)


How to Test:
1. Open the app in a web browser.
2. Enter a Pokémon name or ID in the search input.
3. Click the search button or press "Enter".
4. Verify that the Pokémon details are displayed correctly, including the name, ID, image, types, and stats.
5. Check the responsiveness of the layout on different screen sizes.

pass the freecodecamp test

![pokemon](https://github.com/faridafaqiri/pokemon-search-app/assets/77193878/533093d3-5ae2-4fe3-a7b5-6970c18b4b1b)
